front/src/applications/stdcm/utils/addSecondaryCodesToSimilarTrains.ts

```diff
@@ -0,0 +1,30 @@
+import type { StdcmPathStep } from 'reducers/osrdconf/types';
+
+import type { SimilarTrainWithSecondaryCode } from '../types';
+
+export const addSecondaryCodesToSimilarTrains = (
+  similarTrains: SimilarTrainWithSecondaryCode[],
+  pathSteps: StdcmPathStep[]
+): SimilarTrainWithSecondaryCode[] => {
```

Comment:
When you build your `SimilarTrainWithSecondaryCode`, you know that:
- the train_name is defined
- the secondary_code are defined

So you can modify the types `SimilarTrainWithSecondaryCode` and `WaypointWithSecondaryCode`, and remove the `??` in SimilarTrainsToDuplicate